File: lang/java/test/fun/vac/algorithms/sorting/SelectionSortTest.java
import fun.vac.algorithms.sorting.SelectionSort;
import fun.vac.util.SequenceBuilder;
import fun.vac.util.Sequences;
import fun.vac.util.TestRunner;

public class SelectionSortTest {

    public static boolean testSelectionSort() {
        int size = 32768;

        Integer[] arr = new Integer[size];
        SequenceBuilder.packRandom(arr);

        int checksum = Sequences.parityChecksum(arr);

        SelectionSort.sort(arr);

        if (Sequences.parityChecksum(arr) != checksum) {
            return false;
        }

        return Sequences.isSorted(arr);
    }

    public static void main(String[] args) {
        TestRunner.parseTest(testSelectionSort());
    }
}

File: lang/python/test/fun/vac/algorithms/sorting/SelectionSortTest.py
from fun.vac.algorithms.sorting import SelectionSort
from fun.vac.util import SequenceBuilder
from fun.vac.util import Sequences
from fun.vac.util import TestRunner

def testSelectionSort() -> bool:
    size = 32768

    arr = [0] * size
    SequenceBuilder.packRandom(arr)

    checksum = Sequences.parityChecksum(arr)

    SelectionSort.sort(arr)

    if Sequences.parityChecksum(arr) != checksum:
        return False

    return Sequences.isSorted(arr)

if __name__ == "__main__":
    TestRunner.parseTest(testSelectionSort())
